core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -340,4 +346,26 @@ public static Processor readProcessor(Map<String, Processor.Factory> processorFa
         }
         throw newConfigurationException(type, tag, null, "No processor type exists with name [" + type + "]");
     }
+
+    @SuppressWarnings("unchecked")
+    static Script readOptionalScript(String processorType, String processorTag, Map<String, Object> configuration,
+                                            String propertyName) {
+        try {
+            XContentBuilder builder = XContentBuilder.builder(JsonXContent.jsonXContent);
+            Object scriptObject = configuration.remove(propertyName);
+            if (scriptObject == null) {
+                return null;
+            }
+            if (scriptObject instanceof String) {
+                builder.value(scriptObject);
+            } else {
+                builder.map((Map<String, Object>) scriptObject);
+            }
+            JsonXContentParser parser = new JsonXContentParser(NamedXContentRegistry.EMPTY,
```

Comment:
no specific good reason, just still getting used to the more idiomatic helper libraries for working with xcontent